File: src/main/java/ma/fstg/gestionprojets/services/ProjetService.java
package ma.fstg.gestionprojets.services;
import ma.fstg.gestionprojets.dto.request.ProjetRequest;
import ma.fstg.gestionprojets.dto.response.*;
import java.util.List;
public interface ProjetService {
    ProjetResponse create(ProjetRequest req);
    ProjetResponse update(Long id, ProjetRequest req);
    ProjetResponse getById(Long id);
    List<ProjetResponse> getAll();
    ProjetResumeResponse getResume(Long id);
    List<ProjetResponse> getEnCours();
    List<ProjetResponse> getClotures();
    void delete(Long id);
}

File: gestion-projets-frontend/src/services/projetService.ts
import api from '../utils/axiosConfig';

export const projetService = {
  getAll: () => api.get('/projets'),
  getById: (id: number) => api.get(`/projets/${id}`),
  getResume: (id: number) => api.get(`/projets/${id}/resume`),
  create: (data: any) => api.post('/projets', data),
  update: (id: number, data: any) => api.put(`/projets/${id}`, data),
  delete: (id: number) => api.delete(`/projets/${id}`),
};
